tests/lib/rules/no-shadow-route-definition.js

```diff
@@ -101,6 +101,20 @@ ruleTester.run('no-shadow-route-definition', rule, {
     // With dynamic/variable route or path name:
     'this.route(someVariable);',
     "this.route('views', { path: someVariable })",
+    `this.route(someVariable, function () {
+      this.route('route');
+    });
+    `,
+    `this.route('first', { path: foo });
+    this.route('second', { path: 'foo' });
+    `,
+    `this.route(foo);
+    this.route('foo');
+    `,
+    'this.route(null)',
+    'this.route(true)',
+    // Test not crashing on unexpected values
+    'this.route({})',
 
```

Comment:
Can you add test cases with a function call as the route name / path to make sure we don't crash? Note that I think we should just bail out if we see a function call, we should not try to report linting violations in that situation.

```
this.route(getRouteName());
```

```
this.route('my-route', { path: getRouteName() });
```

Reply:
Will do.

Reply:
Added.

Reply:
Sorry, one last change, can you actually add a string route nested inside these test cases? Otherwise, we might not be fully exercising the code.

```js
this.route(getRouteName(), function () {
  this.route('route');
});
```

```js
this.route("my-route", { path: getRoutePath() }, function () {
  this.route('route');
});
```

Reply:
You are correct, we did miss few bits. Pushed changes to get those examples passing.